api/model-server.yaml

```
@@ -0,0 +1,607 @@
+# WARNING - EXPERIMENTAL
+# This file was auto generated from the existing API using an IntelliJ plugin,
+# see https://www.jetbrains.com/help/idea/openapi.html#generate_openapi
+#
+# Manual changes were done for this 'spec' to work in-place with the
+# model-server for now. A lot of changes were done and are still necessary
+# to make this OpenAPI a viable artifact. It will most likely be split
+# into multiple OpenAPI files.
+
+openapi: "3.0.3"
+info:
+  title: "model-server public API"
+  description: "modelix public API"
+  version: "1.0.0"
+servers:
+  - url: '/'
+    description: model-server
+paths:
+  /v2/server-id:
```

```suggestion
  /server-id:
```

and so on. That way, we can more easily change our mind about the location of the API

---

those will be breaking changes - I did not intend to have a breaking change in this PR but rather just simply transition to resource based paths. I will add a follow up issue in which we can discuss these breaking changes in advance

---

they wouldn't be? Because the base path on top is changed, too?

---

but that change would move all non `/v2` paths, such as `/health` to `/v2/health` and thus create a breaking change?

---

We need to follow up on this. I created an issue here to separate this discussion from this PR.

[MODELIX-683](https://issues.modelix.org/issue/MODELIX-683) Decide and organize model-server endpoints (via OpenAPI definition)